zeebe/gateway-grpc/src/main/java/io/camunda/zeebe/gateway/interceptors/impl/IdentityInterceptor.java

```diff
@@ -30,6 +32,7 @@ public final class IdentityInterceptor implements ServerInterceptor {
   private static final Metadata.Key<String> AUTH_KEY =
       Metadata.Key.of("Authorization", Metadata.ASCII_STRING_MARSHALLER);
   private final Identity identity;
```

Comment:
💭 If we also put the token verification in the wrapper we don't need his field here anymore and it all goes through the wrapper. That way we don't differentiate between sdk calls, which could cause confusion.

Reply:
The problem is that this code is gRPC-specific. My goal with the wrapper was to make it reusable for gRPC and the REST API. I would need to play around with lambdas (maybe even generics) to achieve that. 

It will probably be a nicer solution than what is currently provided, but we need to deliver a fix for this by next week, and I don't want to risk introducing a regression.

@remcowesterhoud would it be acceptable if I create a `kind/toil` issue for this?

Reply:
Maybe I'm missing something, but I thought the `IdentityAuthenticationManager` was the rest part? They both make a call to `identity.authentication().verifyToken(tokenValue);` which seems like we could move it to the wrapper.

Anyway it was only a thought, feel free to ignore it.

Reply:
Ah okay, I misunderstood. I thought you wanted to wrap the token extraction from the request as well.

During the solution discussion with @npepinpe, we decided to avoid adding other `identity-sdk` requests to this wrapper since no problems have been reported about them. We can expand the wrapper if additional issues are reported.

In the end, this fix is only a band-aid. The best solution would be to make Identity more scalable and we are currently making Identity more scalable by integrating its data model in Zeebe. After we're finished with the re-architecture, this fix won't be needed.